include/envoy/grpc/async_client.h

```diff
@@ -100,13 +155,12 @@ template <class RequestType, class ResponseType> class AsyncClient {
    * @return a stream handle or nullptr if no stream could be started. NOTE: In this case
    *         onRemoteClose() has already been called inline. The client owns the stream and
    *         the handle can be used to send more messages or finish the stream. It is expected that
-   *         finish() is invoked by the caller to notify the client that the stream resources may
-   *         be reclaimed.
+   *         closeStream() is invoked by the caller to notify the client that the stream resources
+   *         may be reclaimed.
    */
-  virtual AsyncClientStream<RequestType>*
-  start(const Protobuf::MethodDescriptor& service_method,
-        AsyncClientCallbacks<ResponseType>& callbacks,
-        const Optional<std::chrono::milliseconds>& timeout) PURE;
+  virtual AsyncStream<RequestType>* start(const Protobuf::MethodDescriptor& service_method,
+                                          AsyncStreamCallbacks<ResponseType>& callbacks,
+                                          const Optional<std::chrono::milliseconds>& timeout) PURE;
```

Comment:
Since we are now splitting streaming and unary, can we kill the timeout in the streaming case? IMO it doesn't make any sense. (I know the underlying AsyncClient stuff still has this problem, but we might as well fix it here?)